jupyterhub/templates/proxy/service.yaml

```diff
@@ -44,26 +43,18 @@ spec:
     {{- if $HTTPS }}
     - name: https
       port: 443
-      protocol: TCP
-      {{- if or $manualHTTPS $manualHTTPSwithsecret }}
-      targetPort: 8443
-      {{- else if $offloadHTTPS }}
-      targetPort: 8000
+      {{- if $offloadHTTPS }}
+      targetPort: http
       {{- else }}
-      targetPort: 443
+      targetPort: https
       {{- end }}
       {{- with .Values.proxy.service.nodePorts.https }}
       nodePort: {{ . }}
       {{- end }}
     {{- end }}
     - name: http
       port: 80
-      protocol: TCP
-      {{- if $autoHTTPS }}
-      targetPort: 80
-      {{- else }}
-      targetPort: 8000
-      {{- end }}
+      targetPort: http
```

Comment:
No matter if this goes to the deploy/autohttps or deploy/proxy, both have a port named http even if it is exposed on different port numbers.